packages/common/src/core/traits/IWithDiscussionsSettings.ts

```diff
@@ -0,0 +1,8 @@
+import { IDiscussionsSettings } from "../../discussions";
+
+/**
+ * Discussions-related settings
+ */
+export interface IWithDiscussionsSettings {
```

Comment:
I think we need to define an additional `trait` interface that this interface extends that adds an additional `entitySettingsId` member to the entity data structure. This is because of how we provide default settings values, and the settings id mirrors that of the entity item's id. Implementing an `entitySettingsId` provides us a means to determine whether we're working with default setting values (no record exists, we need to call createSetting) vs existing settings (need to call updateSetting).


```jsx
// src/core/traits/IWithEntitySettings.ts

/**
 * Entity settings
 */
export interface IWithEntitySettings {
  entitySettingsId?: string;
}

```

```jsx
// src/core/traits/IWithDiscussionsSettings.ts
import { IWithEntitySettings } from ".";
import { IDiscussionsSettings } from "../../discussions";

/**
 * Discussions-related entity settings
 */
export interface IWithDiscussionsSettings extends IWithEntitySettings {
  discussionSettings: IDiscussionsSettings;
}

```